src/content/language/type-system.md

```diff
@@ -372,6 +372,79 @@ The return type of the closure is inferred as `int` using upward information.
 Dart uses this return type as upward information when inferring the `map()`
 method's type argument: `<int>`.
 
+#### Inference using bounds
+
+:::version-note
+Inference using bounds requires a [language version][] of at least 3.7.0.
+:::
+
+With the inference using bounds feature,
+Dart's type inference algorithm generates constraints by
+combining existing constraints with the declared type bounds,
+not just best-effort approximations.
+
+This is especially important for [F-bounded][] types,
+where inference using bounds correctly infers that, in the example below,
+`X` can be bound to `B`.
+Without the feature, the type argument must be specified explicitly: `f<B>(C())`:
+
+<?code-excerpt "lib/strong_analysis.dart (inference-using-bounds)"?>
+```dart
+class A<X extends A<X>> {}
+
+class B extends A<B> {}
+
+class C extends B {}
+
+void f<X extends A<X>>(X x) {}
+
+void main() {
+  f(B()); // OK.
+  f(C()); // OK. Without using bounds, inference relying on best-effort
+  // approximations would fail after detecting that `C` is not a subtype of `A<C>`.
+  f<B>(C()); // OK.
+}
+```
+
+Here's a more realistic example using everyday types in Dart like `int` or `num`,
+that typically have their own special properties:
```

Comment:
It might be a good idea to omit the part about the special properties: When I mentioned the special properties it was just because they could be a distraction. At this point we already have an example that clearly doesn't depend on special built-in-type-ish properties, so we just need to avoid the distraction:
```suggestion
Here's a more realistic example using everyday types in Dart like `int` or `num`:
```